src/eins-network-id.c

```diff
@@ -0,0 +1,238 @@
+/* Copyright 2017 Endless Mobile, Inc.
+ *
+ * This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <stdio.h>
+
+#include "eins-network-id.h"
+
+#define PROC_IPV4_ROUTE_FILE "/proc/net/route"
```

Comment:
That's a good Q, although given the number of architectures in the universe is trending down with the exception of backwards-compatibility with two little-endian mainstays, may not be all that critical for Endless. For the sake of completeness I'll check it out. This file definitely has the IPs as little-endian, but I don't know if that's because it's host-endian or because the trie has them backwards for some implementation reason. I don't want to stick it through htonl only to find it comes out in the same order on big-endian machines too! :D